### ISSUE TITLE
4.0 beta version created groups are deleted after reopen

### ISSUE THREAD

**Issue by mishrabk** [JabRef/jabref#2912]
JabRef version  on 

Steps to reproduce:

1. ...
2. ...
3. ...

<details>
  <summary>Log File</summary>
  
  ```
  Paste an excerpt of your log file here
  ```
</details>

**Comment by Siedlerchr** [comment]
Please try again with the latest snapshot

**Comment by mishrabk** [comment]
Tried (JabRef_windows-x64_4_0_0-dev--snapshot--2017-06-15--master--0c006c1e1), but still its not working same problem.

**Comment by AEgit** [comment]
Cannot confirm for:
JabRef 4.0.0-dev--snapshot--2017-06-06--master--5dcdd79ab
Windows 10 10.0 amd64 
Java 1.8.0_131

What happens, however, is that the database is not marked as 'dirty' when a new group is created. This has already been reported here: https://github.com/JabRef/jabref/issues/2787
If you don't save your database manually after creating a new group, it won't be saved at all, when you close the database (i.e., the changes to the database won't be saved automatically).

I'll try the current developer version as well, but I guess things stay the same.

**Comment by AEgit** [comment]
As expected, I can't confirm the problem for:
JabRef 4.0.0-dev--snapshot--2017-06-15--master--0c006c1e1
Windows 10 10.0 amd64 
Java 1.8.0_131

@mishrabk : Probably you don't save your database after having created the new group - that's why it appears as if the groups were deleted when you reopen the database. What actually happens is that the groups have never been stored (if you don't trigger the saving mechanism manually).

**Comment by mishrabk** [comment]
@AEgit: I am saving database manually.

I am using windows 7x64
java 1.8.0_91-b14

**Comment by tobiasdiez** [comment]
Can you please tell us the exact steps to reproduce this problem. Does it occur always or just sometimes? After you save and close the file, does the `jabref-comment: group` comment at the end of the bib file contains the added group? Do you see anything in the error log?

**Comment by AEgit** [comment]
Sorry for the obvious question (I just want to make sure, I understand your problem), but what exactly happens, when you reopen your database?
Is the group interface just not shown (you can toggle it on using `ALT+3`)?
Or have the groups that you created initially actually been deleted? If the latter is the case, did you follow the following procedure (?):

1. Open JabRef database.
2. Open groups interface with `ALT+3`. Add a new group/subgroup (I presume you add a static group?).
3. Save the JabRef database using `CTRL + S`.
4. Close the JabRef database.
5. Reopen the JabRef database.
6. Open groups interface with `ALT+3`. The groups you created should still be present in the database.

If you have followed all these steps, but the groups have actually disappeared, I can only speculate.
First of all, what is your JabRef version? Using `Help -> About JabRef -> Copy Version` gives you the version of your JabRef installation.
Second, your Java Runtime Environment is not up to date. You are using `Java 1.8.0_91-b14`, but the current one is `Java 1.8.0_131`.
Third, it could be a difference between Windows 7 and Windows 10 (I think that's unlikely, but who knows).
As far as I can tell, these are the only differences between our systems, so I guess one of them should explain, why your groups are not saved, while mine are.

**Comment by mishrabk** [comment]
@tobiasdiez @AEgit : I do hope I would be able to convey what exactly I am doing.

1. Open JabRef database
2. Open groups interface with ALT+3 or by clicking the icon
3. Click on '+' icon to create a new group (static or dynamic it does not matter)
4. Save the JabRef database
5. Close the JabRef database.
6. Reopen the JabRef database.
7. The group I created is no more in the database.

I updated my java environment now I have:

java: 1.8.0_131-b11, windows 7x64, jabref: JabRef_windows-x64_4_0_0-dev--snapshot--2017-06-15--master--0c006c1e1.

@tobiasdiez : For me it happens all the time.

No, (jabref-comment: group) comment at the end of the bib file does not contains the added group.

No, I did not found anything. This could be because, I do not know how I can see complete error log. From help>view event log, it gives some log thent writes ..xx more.

One more thing I did not found create new group under Groups menu. see attached screen shot:
![image](https://user-images.githubusercontent.com/29423115/27252727-27c5fada-5383-11e7-868f-4b7c494c565f.png)

**Comment by AEgit** [comment]
Hmmm, this is very strange - I cannot reproduce this behaviour with:
```
JabRef 4.0.0-dev--snapshot--2017-06-15--master--0c006c1e1
Windows 10 10.0 amd64 
Java 1.8.0_131
```

You have the same JabRef version and the same Java environment. What differs is, that you are using Windows 7, while I'm running Windows 10. But I still don't think, that is the issue (although it would be nice, if someone else with Windows 7 could check). Maybe some of the JabRef settings for groups are different, but I wouldn't know, which ones that could be (?).

BTW: If you want to create a subgroup for another group you can just rightclick on any group in the groups panel. A menu opens which allows you to create a subgroup using "Add subgroup".

**Comment by Yery** [comment]
@mishrabk  Tried it here, but could also not reproduce your problem (also win10). When you close jabref and you look at the "modified date" of the bib file in your file explorer, does it actually show the right time (time of saving?).; in other words did Jabref really save the file?

**Comment by tobiasdiez** [comment]
@mishrabk Thanks for the details. Sadly I also cannot reproduce the problem. If the `jabref` comment at the end of the bib file is missing then the issue results from a problem while saving. Are other changes (e.g. to entries) correctly saved?

**Comment by mishrabk** [comment]
Yes, the group information remains missing from comment section. However, when I tried with jabref 3.8.2 thing are working. When the file created by jabref 3.8.2 is opened in JabRef_windows-x64_4_0_0-dev--snapshot--2017-06-15--master--0c006c1e1 the created groups are not deleted. Also now I can create a group in JabRef_windows-x64_4_0_0-dev--snapshot--2017-06-15--master--0c006c1e1 but doing this caused another problem for me. The group name (keyword for dynamic group is same as group name) got added in the title of each entry.

I agree with your diagnosis i.e. problem while saving. Because bib file does not change (i mean comment section remains unaffected) if I mark/unmark entries.
